### PR TITLE
Bluetooth: controller: Fix network privacy with resolution disabled

### LINKED ISSUE #24731: Bluetooth: controller: Network privacy not respected when address resolution is disabled
**Describe the bug**
The resolving list is not used when address resolution has been disabled by the host.
The address resolution enable command should (after a few erratas) only control resolving of RPAs.
If a device has been added to the resolving list with a valid IRK, and address resolution is not enabled. This should act as a blacklist, and we should not connect to it, or accept connection requests from it.
Since we cannot resolve it, and using it's identity is a violation of network privacy.

**To Reproduce**
Uncomment the following line in `tests/bluetooth/bsim_bt/edtt_ble_test_app/tests_scripts/ll.set1.test_list` and run it according to the readme:
`#LL/CON/INI/BV-24-C #currently failing, to be investigated`

**Expected behavior**
Initiator should not connect to devices in the resolving list with a valid IRK when address resolution is disabled.

**Impact**
Failing LL conformance testcase.
Affects all vendors of Split, and Legacy controller.

**Additional information**
The test has a TSE waiver, but that is not related to why this test is failing now. But would explain why we have qualified without this test before:
https://www.bluetooth.org/tse/errata_view.cfm?errata_id=11429
The waiver seems not to apply anymore when using TCRL 2019-2 or higher.

### PR DESCRIPTION
**Pull request by cvinayak**
When address resolution is disabled, an identity address has
been added into the resolving list with peer IRK, and device
privacy has not been selected for the peer device then
connection indication shall not be sent to the peer that is
advertising using its identity address.

Fixes #24731.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by erbr-ot — APPROVED** [review]
LGTM

**Review by jhedberg — APPROVED** [review]

**Review by George-Stefan — APPROVED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_filter.c [hunk 1]
```diff
@@ -757,10 +757,10 @@ bool ull_filter_lll_rl_addr_allowed(uint8_t id_addr_type, uint8_t *id_addr, uint
 {
 	uint8_t i, j;
 
-	/* If AR is disabled or we matched an IRK then we're all set. No hw
+	/* We matched an IRK then we're all set. No hw
 	 * filters are used in this case.
 	 */
-	if (!rl_enable || *rl_idx != FILTER_IDX_NONE) {
```
Comment (joerchan) [inline subsys/bluetooth/controller/ll_sw/ull_filter.c:763]: By removing the `!rl_enable` here it means that the address comparison below can be with an entry that is in the process of being updated. When adding an RL entry ll_rl_add calls rl_access_check always allows modifications of the RL list when address resolutios is disabled, even if the adv/scan role is active.

The check below can therefore be under modification through ll_rl_add / ll_rl_remove / ll_priv_mode_set, to me this looks like it is safe, but would be nice if @carlescufi and @cvinayak could confirm this.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c:763]: rl_access_check is called with check_ar parameter set to false in the ll_rl_add/remove, which is spec requirement that the resolving list not be allowed to be modified, in this case, when the initiator is active.

Comment (joerchan) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c:763]:
```
This command may be used at any time when address resolution is disabled in
the Controller.
```
It's different with address resolution enabled. I'm unsure if there is a test that actually checks that you can modify the RL with initiator active, and address resolution disabled, but AFAIK that should be allowed.

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c:763]: If `rl_access_check` calls needs to be fixed, and if so done then the `ull_filter_lll_rl_addr_allowed` call from LLL is still safe, as the `taken` flag is checked before accessing the members of the resolving list data structure. `taken` ensures there is sane data members in the resolving list data structures. (may be need consideration of memory barrier instructions though).

In the future if a conformance test is added, or a user faces an issue, we can address the current implementation limitation of not allowing updates to resolving list, then.

Comment (joerchan) [reply subsys/bluetooth/controller/ll_sw/ull_filter.c:763]: Yeah, that was my conclusion as well.

**Pull request merged**